### PR TITLE
Exercise 1

### PR DESCRIPTION
**Pull request by oskarwilliams378**
Did the stretch goals except the styling one

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Jonesey13) [inline .idea/misc.xml:1]: Most of these files should probably be in the .gitignore

File: app.py [hunk 2]
```diff
@@ -1,12 +1,43 @@
...
+
+@app.route('/AddItem/Save', methods=['POST'])
+def add_item_save():
+    title = request.form.get('title')
+    print(title)
+    add_item(title)
+    return redirect('/')
+
+
+@app.route('/EditItem', methods=['POST'])
```
Comment (Jonesey13) [inline app.py:28]: This is probably worth splitting up into two routes

File: session_items.py [hunk 3]
```diff
@@ -67,3 +67,19 @@ def save_item(item):
...
+def remove_item(item_id):
+    """
+    Removes an existing item from the session. If no existing item matches the ID of the specified item, nothing is
+    deleted.
+
+    Args:
+        item_id: The id of the item to remove.
+    """
+    existing_items = get_items()
+    updated_items = [existing_item for existing_item in existing_items if not (int(item_id) == existing_item['id'])]
```
Comment (Jonesey13) [inline session_items.py:81]: Could use != here (good use of a list comprehension btw!)

File: app.py [hunk 4]
```diff
@@ -1,12 +1,43 @@
...
+
+
+@app.route('/AddItem')
+def add_item_page():
+    return render_template('addItem.html')
+
+
+@app.route('/AddItem/Save', methods=['POST'])
+def add_item_save():
+    title = request.form.get('title')
```
Comment (Jonesey13) [inline app.py:22]: Could probably add some server side validation here

File: .idea/.gitignore [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
```
[the remark this answers is not in the document]
Comment (oskarwilliams378) [reply .idea/.gitignore:1]: Out of interest, why vscode over pycharm?

[the remark this answers is not in the document]
Comment (Jonesey13) [reply .idea/.gitignore:1]: The main reason is probably consistency as we'll be using vscode live share for the workshops going forward. Some other reasons:
* Free (so don't have to worry about external companies buying licenses)
* Lightweight and modular (makes it harder for people to dig themselves into a configuration hole; easier for us to load and run projects)
* Lack of any particular advantages for using with Python (that I'm aware of) unlike Java and C# where the IDE support is more advanced.

[the remark this answers is not in the document]
Comment (oskarwilliams378) [reply .idea/.gitignore:1]: Ahh ok. I will always check against vscode then and make sure to keep that working easily. AFAIK Pycharm community edition (the free one) does actually still provide a lot of Python specific IDE support which I have enjoyed in the past.

**Pull request merged**